apps/backend/src/app/bids/bids.service.ts

```diff
@@ -18,44 +18,49 @@ export class BidsService {
     private auctionsRepository: Repository<Auction>
   ) {}
 
-  maxBid(auctionId: string) {
+  maxBid(auctionId: string): Promise<Bid> {
     const byAuctionId = findByAuctionId(auctionId);
 
     return this.bidsRepository.findOne(byAuctionId);
   }
 
   async create(createBidDto: CreateBidDto) {
-    const auction = await this.auctionsRepository.findOne({
-      where: {
-        id: createBidDto.auction,
-      },
-    });
+    const queryRunner = this.bidsRepository.manager.connection.createQueryRunner();
+    await queryRunner.startTransaction();
 
-    const user = await this.usersRepository.findOne({
-      where: {
-        id: createBidDto.user,
-      },
-    });
+    try {
+      const auction = await this.auctionsRepository.findOneOrFail({
+        where: { id: createBidDto.auction },
+      });
 
-    if (!auction || !user) {
-      throw new HttpException('Bad request', HttpStatus.BAD_REQUEST);
-    }
+      const user = await this.usersRepository.findOneOrFail({
```

Comment:
@pawelroguski this will not return 400 bad request. Thats why i used findOne.

Reply:
In case we use findOneOrFail we will get EntityNotFoundError witch is definetly much better message for this example. I think in this case if statement is not necessary

Reply:
@pawelroguski But API should return 400 bad request. Because request was bad :)
You should then change `catch` so it will return bad request if error type is `EntityNotFoundError `

Reply:
Yes I added it, You can check ;)

Reply:
@pawelroguski i checked it and got 
```{
  "statusCode": 404,
  "message": "Not Found"
}```
right now im very confused. 
![image](https://user-images.githubusercontent.com/82712780/232055046-69c34e60-4d3c-428d-ba42-8d6807552065.png)

Reply:
Me too because in any case I don't have response like yours ;) On wich branch You checked this functionality. Are You sure that was INHEART-447. If it was correct branch, maybe You didn't pull it ?